packages/core/contracts/gnosis-zodiac/implementation/OptimisticGovernor.sol

```diff
@@ -215,39 +241,41 @@ contract OptimisticGovernor is Module, Lockable {
         // Create the proposal hash.
         bytes32 proposalHash = keccak256(abi.encode(_transactions));
 
-        // Add the proposal hash to ancillary data.
-        bytes memory ancillaryData = AncillaryData.appendKeyValueBytes32("", PROPOSAL_HASH_KEY, proposalHash);
+        // Add the proposal hash and explanation to ancillary data.
+        bytes memory claim = _constructClaim(proposalHash, _explanation);
 
-        // Check that the proposal is not already mapped to a proposal time, i.e., is not a duplicate.
-        require(proposalHashes[proposalHash] == 0, "Duplicate proposals are not allowed");
+        // Check that the proposal is not already mapped to an assertionId, i.e., is not a duplicate.
+        require(proposalHashes[proposalHash] == bytes32(0), "Duplicate proposals not allowed");
 
-        // Map the proposal hash to the current time.
-        proposalHashes[proposalHash] = time;
+        // Check the minimum required bond and use that instead if it is greater than the bondAmount.
+        uint256 minimumBond = optimisticAsserter.getMinimumBond(address(collateral));
+        uint256 totalBond = minimumBond > bondAmount ? minimumBond : bondAmount;
```

Comment:
Do you think it could make sense having a public function that does this totalBond calculation based on the minimumBond so it's easier for consumer contracts to get this number and approve the contract?
Something like `getProposalBond()` that would serve the same purpose as [gitMinimumBond](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/optimistic-asserter/implementation/OptimisticAsserter.sol#L360)

Reply:
good idea, will add this!